src/TorchVision/Utils.cs

```diff
@@ -0,0 +1,132 @@
+                }
+
+                if (normalize == true)
+                {
+                    void norm_ip(Tensor img, double low, double high)
+                    {
+                        img.clamp_(min: low, max: high);
+                        img.sub_(low).div_(Math.Max(high - low, 1e-5));
+                    }
+
+                    void norm_range(Tensor t, (double low, double high)? range)
+                    {
+                        if (range.HasValue)
+                        {
+                            var (low, high) = value_range.Value;
+                            norm_ip(t, low, high);
+                        }
+                        else
+                        {
+                            norm_ip(t, t.min().ToSingle(), t.max().ToSingle());
+                        }
+                    }
+
+                    if (scale_each == true)
+                    {
+                        for (long i = 0; i < tensor.size(0); ++i)
+                        {
+                            norm_range(tensor[i], value_range);
+                        }
+                    }
+                    else
+                    {
+                        norm_range(tensor, value_range);
+                    }
+                }
+
+                if (tensor.size(0) == 1)
+                    return tensor.unsqueeze(0);
+
+                var nmaps = tensor.size(0);
+                var xmaps = Math.Min(nrow, nmaps);
+                var ymaps = (long)Math.Ceiling((double)nmaps / xmaps);
+                var width = tensor.size(3) + padding;
+                var height = tensor.size(2) + padding;
+                var num_channels = tensor.size(1);
+
+                var grid = tensor.new_full(new[] { num_channels, height * ymaps + padding, width * xmaps + padding }, pad_value);
+                var k = 0L;
+                for (long y = 0; y < ymaps; ++y)
+                {
+                    for (long x = 0; x < xmaps; ++x)
+                    {
+                        if (k > nmaps) break;
+                        grid.narrow(1, y * height, height - padding).narrow(
+                            2, x * width + padding, width - padding
+                            ).copy_(tensor[k]);
+                        ++k;
+                    }
+                }
+
+                return grid;
+            }
+
+            public static void save_image(
+                Tensor tensor,
+                string filename,
+                ImageFormat format,
+                long nrow = 8,
+                int padding = 2,
+                bool normalize = false,
+                (double low, double high)? value_range = null,
+                bool scale_each = false,
+                double pad_value = 0.0f,
+                Imager imager = null)
+            {
+                var grid = make_grid(tensor, nrow, padding, normalize, value_range, scale_each, pad_value);
```

Comment:
@NiklasGustafsson If I understand TorchSharp memory management correctly these should be `using var grid` and `using var narr` ?

Reply:
Actually, there should be a dispose scope around the function body. There are temporaries being created that will not be disposed, even with a 'using'. Good catch!

Reply:
Isn't it just `mul()` that creates a tempory? `add_`, `clamp_` and `to` are inplace ops?

Reply:
Yeah, I guess so. Maybe mul_() would be useful here? I can't think of any reason why it couldn't be used.

Reply:
I think I will add a memory unit test:

``c#
using var image = torch.randn(32,3,32,32);
using (var d = torch.NewDisposeScope()) {
    torchvision.utils.save_image(image, ...
    Assert.Equal(d.DisposablesCount, 0);
}
```

That is to say I expect the save_image function to clean up all of its temporaries.

I also expect make_grid to only return one temporary.

Reply:
Yes, that makes sense.

Reply:
> By the way, any unit test that writes to a file should make sure to delete it afterwards.

Or use mocks, like MemoryStream.